private/private.h

```diff
@@ -39,7 +39,7 @@
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#if HAVE_SYS_CDEFS_H
+#if __has_include(<sys/cdefs.h>)
```

Comment:
one concern with this approach here we've always had is that __has_include is not supported on all compilers we support at least in the Darwin SDK (that is why dispatch/base.h defines them). 

I would suggest replicating the approach at the top of dispatch/dispatch.h here (replacing lines 30-32)

```
#ifdef __APPLE__
#include <Availability.h>
#include <os/availability.h>
#include <TargetConditionals.h>
#include <os/base.h>
#elif defined(__linux__)
#include <os/linux_base.h>
#endif
```

and then actually dropping the `#include <sys/cdefs.h>` completely here (as it will already be covered by the base*.h includes)